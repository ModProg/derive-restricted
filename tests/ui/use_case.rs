use derive_where::derive_where;

#[derive_where(Clone)]
struct StructEmpty {}

#[derive_where(Clone)]
struct TupleEmpty();

#[derive_where(Clone)]
struct Unit;

#[derive_where(Clone)]
struct UnionEmpty {}

#[derive_where(Clone)]
struct StructNone(u8);

#[derive_where(Clone)]
enum EnumNone {
	A(u8),
}

#[derive_where(Clone; T)]
struct SameGenerics<T>(T);

#[derive_where(Clone, Debug; T)]
#[derive_where(skip_inner)]
struct Skip<T>(T);

#[derive_where(Clone, Default; T)]
enum Default<T> {
	#[derive_where(default)]
	A(T),
}

fn main() {}
