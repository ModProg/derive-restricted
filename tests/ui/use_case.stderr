error: derive-where doesn't support empty items, as this can already be handled by standard `#[derive(..)]`
 --> tests/ui/use_case.rs:4:1
  |
4 | struct StructEmpty {}
  | ^^^^^^^^^^^^^^^^^^^^^

error: derive-where doesn't support empty items, as this can already be handled by standard `#[derive(..)]`
 --> tests/ui/use_case.rs:7:1
  |
7 | struct TupleEmpty();
  | ^^^^^^^^^^^^^^^^^^^^

error: derive-where doesn't support empty items, as this can already be handled by standard `#[derive(..)]`
  --> tests/ui/use_case.rs:10:1
   |
10 | struct Unit;
   | ^^^^^^^^^^^^

error: derive-where doesn't support empty items, as this can already be handled by standard `#[derive(..)]`
  --> tests/ui/use_case.rs:13:1
   |
13 | struct UnionEmpty {}
   | ^^^^^^^^^^^^^^^^^^^^

error: this can be handled by standard `#[derive(..)]`, use a `skip` attribute, implement `Default` on an enum, of different generic type parameters
  --> tests/ui/use_case.rs:15:16
   |
15 | #[derive_where(Clone)]
   |                ^^^^^

error: this can be handled by standard `#[derive(..)]`, use a `skip` attribute, implement `Default` on an enum, of different generic type parameters
  --> tests/ui/use_case.rs:18:16
   |
18 | #[derive_where(Clone)]
   |                ^^^^^

error: this can be handled by standard `#[derive(..)]`, use a `skip` attribute, implement `Default` on an enum, of different generic type parameters
  --> tests/ui/use_case.rs:23:16
   |
23 | #[derive_where(Clone; T)]
   |                ^^^^^

error: this can be handled by standard `#[derive(..)]`, use a `skip` attribute, implement `Default` on an enum, of different generic type parameters
  --> tests/ui/use_case.rs:26:16
   |
26 | #[derive_where(Clone, Debug; T)]
   |                ^^^^^

error: this can be handled by standard `#[derive(..)]`, use a `skip` attribute, implement `Default` on an enum, of different generic type parameters
  --> tests/ui/use_case.rs:30:16
   |
30 | #[derive_where(Clone, Default; T)]
   |                ^^^^^
